src/Entity/User.php

```diff
@@ -435,33 +435,20 @@ public function removeVote(Vote $vote): self
         return $this;
     }
 
-    public function hasVotedBug(Bug $bug): bool
+    public function hasVoted(UserRequest $request): bool
     {
-        return $this->votes->exists(fn (int $key, Vote $vote) => $vote->getBug() === $bug);
+        return null !== $this->getVote($request);
     }
 
-    public function getVoteForBug(Bug $bug): ?Vote
+    public function getVote(UserRequest $request): ?Vote
     {
-        $vote = $this->votes->filter(fn (Vote $vote) => $vote->getBug() === $bug)->first();
-
-        return false === $vote ? null : $vote;
-    }
-
-    public function getVoteForFeature(Feature $feature): ?Vote
-    {
-        $vote = $this->votes->filter(fn (Vote $vote) => $vote->getFeature() === $feature)->first();
-
-        return false === $vote ? null : $vote;
-    }
-
-    public function getVoteForItem(UserRequest $item): ?Vote
-    {
-        if ($item instanceof Feature) {
-            return $this->getVoteForFeature($item);
-        } elseif ($item instanceof Bug) {
-            return $this->getVoteForBug($item);
+        $vote = false;
+        if ($request->isFeature()) {
+            $vote = $this->votes->filter(fn (Vote $vote) => $vote->getBug() === $request)->first();
+        } elseif ($request->isBug()) {
+            $vote = $this->votes->filter(fn (Vote $vote) => $vote->getFeature() === $request)->first();
         }
 
-        return null;
+        return false === $vote ? null : $vote;
```

Comment:
We can use elvis operator here `return $vote ?: null;`

Reply:
Also, maybe we could add method `$vote->getItem(): UserRequest` (as `vote->setItem()` already exists).
And simplify this as :
`return $this->votes->filter(fn (Vote $vote) => $vote->getBug() === $request)->first() ?: null;`